src/spartan/sumcheck.rs

```diff
@@ -105,21 +105,43 @@ impl<G: Group> SumcheckProof<G> {
   {
     let len = poly_A.len() / 2;
     (0..len)
-      .into_par_iter()
-      .map(|i| {
-        // eval 0: bound_func is A(low)
-        let eval_point_0 = comb_func(&poly_A[i], &poly_B[i]);
-
-        // eval 2: bound_func is -A(low) + 2*A(high)
-        let poly_A_bound_point = poly_A[len + i] + poly_A[len + i] - poly_A[i];
-        let poly_B_bound_point = poly_B[len + i] + poly_B[len + i] - poly_B[i];
-        let eval_point_2 = comb_func(&poly_A_bound_point, &poly_B_bound_point);
-        (eval_point_0, eval_point_2)
-      })
-      .reduce(
-        || (G::Scalar::ZERO, G::Scalar::ZERO),
-        |a, b| (a.0 + b.0, a.1 + b.1),
-      )
+            .into_par_iter()
+            .map(|i| {
+                // This `map` function computes point-wise combinations (via `comb_func`) of the evals of each
+                // polynomial -- as though the 'top var' (next to be bound in sumcheck) have been bound to the target.
+                //
+                // The target is the evaluation that will be used to create a univariate-polynomial whose evaluation at
+                // the target is the sum of the combined non-top evaluations.
+                //
+                // The following `reduce `operation effects the sum.
+                //
+                // The required targets are the evaluation points 0 and 2.
+                //
+                // Evaluation at 0 is unaffected by the top var (to be bound).
+                // eval 0: bound_func is A(low)
```

Comment:
This comment is confusing. I believe I understand what you're getting at, but I think an explanation like this could be clearer:

> This calculates the univariate polynomial obtained by summing over all but the top var.
> The univariate polynomial is specified by its evaluations at 0, and 2 which are obtained by combining the evaluations of each multivariate polynomial at 0 and 2 with `comb_func`. 

It also may be worth mentioning why only the evaluations of 0 and 2 are needed (quadratic polynomial + 1 can be derived from the hint)